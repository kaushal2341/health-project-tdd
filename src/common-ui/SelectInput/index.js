import React from 'react';
import  {Dropdown} from 'semantic-ui-react';
export default class SelectInput extends React.PureComponent {
   onChangeHandler = (e , data) => {
     const myEvent = {...e}
     myEvent.target.value = e.target.value ? e.target.value : data.value
     myEvent.target.name = data.name
     this.props.onChangeHandler( myEvent );
   }       
   render(){
     return(
      <>
      <Dropdown
       name={this.props.name}
       additionLabel={this.props.additionLabel}
       additionPosition={this.props.additionPosition}
       allowAdditions={this.props.allowAdditions}
       as={this.props.as}
       basic={this.props.basic}
       button={this.props.button}
       children={this.props.children}
       className={this.props.className}
       clearable={this.props.clearable}
       closeOnChange={this.props.closeOnChange}
       compact={this.props.compact}
       deburr={this.props.deburr}
       defaultOpen={this.props.defaultOpen}
       defaultUpward={this.props.defaultUpward}
       defaultValue={this.props.defaultValue}
       icon={this.props.icon}
       direction={this.props.icon}
       disabled={this.props.disabled}
       error={this.props.error}
       floating={this.props.floating}
       fluid= {this.props.flluid}
       header={this.props.header} 
       inline={this.props.inline}
       item={this.props.item}
       labeled={this.props.labeled}
       lazyLoad={this.props.lazyLoad}
       loading={this.props.loading}
       minCharacters={this.props.minCharacters}
       multiple={this.props.multiple}
       noResultsMessage={this.props.noResultsMessage}
       onAddItem={this.props.onAddItem}
       onBlur={this.props.onBlur}
       onChange={(event,data) => this.onChangeHandler(event,data)}
       onClick={this.props.onClick}
       onClose={this.props.onClose}
       onFocus={this.props.onFocus}
       onLabelClick={this.props.onLabelClick}
       onMouseDown={this.props.onMousedDown}
       onOpen={this.props.onOpen}
       onSearchChange={this.props.onSearchChange}
       open={this.props.open}
       openOnFocus={this.props.openOnFocus}
       options={this.props.options}
       placeholder={this.props.palceholder}
       pointing={this.props.pointing}
       renderLabel={this.props.renderLabel}
       scrolling ={this.props.scrolling}
       search={this.props.search}
       searchInput={this.props.searchInput}
       searchQuery={this.props.searchQuery}
       selectOnBlur={this.props.selectOnBlur}
       selectOnNavigation={this.props.selectOnNavigation}
       selectedLabel={this.props.selectedLabel}
       selection={this.props.selection}
       simple={this.props.simple}
       tabIndex={this.props.tabIndex}
       text={this.props.text}
       trigger={this.props.trigger}
       upward={this.props.upward}
       value={this.props.value}
       wrapSelection={this.props.wrapSelection}
       closeOnBlur={this.props.closeOnBlur}
      />
     </>
    )
    }
} 