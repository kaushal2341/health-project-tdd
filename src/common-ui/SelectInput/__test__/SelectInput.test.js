import React from 'react';
import  {Dropdown} from 'semantic-ui-react';
import SelectInput from  '..';
expect.addSnapshotSerializer(enzymeSerializer);
describe('SelectInput Component', () => {
    let wrapper,onChangeMock;
    beforeEach(() => {
    onChangeMock = jest.fn();    
    wrapper = mount(<SelectInput 
                    value="hello" 
                    options={[{key:'hello',text:'me',value:'i'},{key:'hi',text:'you',value:'yours'}]}
                    onChangeHandler={onChangeMock}
                    />
                   )
    })
    it('should be defined and render successfully', () => {
     expect(wrapper).toBeDefined();
    });
    it('should have only one select input tag', () => {
     expect(wrapper.find(Dropdown)).toHaveLength(1); 
    });
    it('should have the prop options available', () => {
        console.log(wrapper.prop('value'))
    expect(wrapper.prop('options')).toBeDefined();
    });
    it('should have the properties of options equal to key text value',() => {
    expect(wrapper.prop('options')).toEqual([{key:'hello',text:'me',value:'i'},{key:'hi',text:'you',value:'yours'}]);    
    })
    it('should have the given props available',() => {
    const propsToCheck =['name',
                         'additionLabel',
                         'additionPosition',
                         'allowAdditions',
                         'as',
                         'basic',
                         'button',
                         'children',
                         'className',
                         'multiple',
                         'clearable',
                         'closeOnChange',
                         'compact',
                         'deburr',
                         'open',
                         'defaultOpen',
                         'defaultUpward',
                         'defaultValue',
                         'direction',
                         'disabled',
                         'error',
                         'floating',
                         'fluid',
                         'header',
                         'icon',
                         'inline',
                         'item',
                         'labeled',
                         'lazyLoad',
                         'loading',
                         'minCharacters',
                         'noResultsMessage',
                         'onAddItem',
                         'onBlur',
                         'onChange',
                         'onClick',
                         'onClose',
                         'onFocus',
                         'onLabelClick',
                         'onMouseDown',
                         'onOpen',
                         'onSearchChange',
                         'openOnFocus',
                         'options',
                         'placeholder',
                         'pointing',
                         'renderLabel',
                         'scrolling',
                         'search',
                         'searchInput',
                         'searchQuery',
                         'selectOnBlur',
                         'selectOnNavigation',
                         'selectedLabel',
                         'selection',
                         'simple',
                         'tabIndex',
                         'text',
                         'trigger',
                         'upward',
                         'value',
                         'wrapSelection',
                         'closeOnBlur'
                        ];
    let propsOfSelect = Object.keys(wrapper.find(Dropdown).props());
    propsToCheck.map((propCheck , i) => {
        expect(propsToCheck).toContain(propsOfSelect[i]);
    });
    });
describe('Select Input Simulation ' , () => {
let changeCallback;    
beforeEach(()=>{
    changeCallback = jest.spyOn(wrapper.instance(),'onChangeHandler');
    wrapper.instance().forceUpdate(); 
    wrapper.find(Dropdown).simulate('change');
}); 

it('should call the function onChangeHandler when input is changed', () => {
    expect(changeCallback).toHaveBeenCalled();
});

it('should call the props function onChangeHandler when input is changed', () => {
    expect(onChangeMock).toHaveBeenCalled();
});
});    

describe('Snap Shot Testing',() => {
        it('should create or match previous snapshot of the component',() => {
            expect(wrapper).toMatchSnapshot(); 
        });
    })
})