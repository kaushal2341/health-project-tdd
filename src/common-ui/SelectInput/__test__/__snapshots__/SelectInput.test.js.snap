// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput Component Snap Shot Testing should create or match previous snapshot of the component 1`] = `
<SelectInput
  onChangeHandler={[MockFunction]}
  options={
    Array [
      Object {
        "key": "hello",
        "text": "me",
        "value": "i",
      },
      Object {
        "key": "hi",
        "text": "you",
        "value": "yours",
      },
    ]
  }
  value="hello"
>
  <Dropdown
    additionLabel="Add "
    additionPosition="top"
    closeOnBlur={true}
    deburr={false}
    icon="dropdown"
    minCharacters={1}
    noResultsMessage="No results found."
    onChange={[Function]}
    openOnFocus={true}
    options={
      Array [
        Object {
          "key": "hello",
          "text": "me",
          "value": "i",
        },
        Object {
          "key": "hi",
          "text": "you",
          "value": "yours",
        },
      ]
    }
    renderLabel={[Function]}
    searchInput="text"
    selectOnBlur={true}
    selectOnNavigation={true}
    value="hello"
    wrapSelection={true}
  >
    <Ref
      innerRef={
        Object {
          "current": <div
            aria-expanded="false"
            class="ui dropdown"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-live="polite"
              class="text"
              role="alert"
            />
            <i
              aria-hidden="true"
              class="dropdown icon"
            />
            <div
              class="menu transition"
            >
              <div
                aria-checked="false"
                aria-selected="true"
                class="selected item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  me
                </span>
              </div>
              <div
                aria-checked="false"
                aria-selected="false"
                class="item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  you
                </span>
              </div>
            </div>
          </div>,
        }
      }
    >
      <RefFindNode
        innerRef={
          Object {
            "current": <div
              aria-expanded="false"
              class="ui dropdown"
              role="listbox"
              tabindex="0"
            >
              <div
                aria-live="polite"
                class="text"
                role="alert"
              />
              <i
                aria-hidden="true"
                class="dropdown icon"
              />
              <div
                class="menu transition"
              >
                <div
                  aria-checked="false"
                  aria-selected="true"
                  class="selected item"
                  role="option"
                  style="pointer-events: all;"
                >
                  <span
                    class="text"
                  >
                    me
                  </span>
                </div>
                <div
                  aria-checked="false"
                  aria-selected="false"
                  class="item"
                  role="option"
                  style="pointer-events: all;"
                >
                  <span
                    class="text"
                  >
                    you
                  </span>
                </div>
              </div>
            </div>,
          }
        }
      >
        <div
          aria-expanded={false}
          className="ui dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          tabIndex={0}
        >
          <div
            aria-live="polite"
            className="text"
            role="alert"
          />
          <Icon
            as="i"
            className=""
            name="dropdown"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="dropdown icon"
              onClick={[Function]}
            />
          </Icon>
          <DropdownMenu>
            <div
              className="menu transition"
            >
              <DropdownItem
                active={false}
                key="hello"
                onClick={[Function]}
                selected={true}
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }
                text="me"
                value="i"
              >
                <div
                  aria-checked={false}
                  aria-selected={true}
                  className="selected item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    me
                  </span>
                </div>
              </DropdownItem>
              <DropdownItem
                active={false}
                key="hi"
                onClick={[Function]}
                selected={false}
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }
                text="you"
                value="yours"
              >
                <div
                  aria-checked={false}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    you
                  </span>
                </div>
              </DropdownItem>
            </div>
          </DropdownMenu>
        </div>
      </RefFindNode>
    </Ref>
  </Dropdown>
</SelectInput>
`;
